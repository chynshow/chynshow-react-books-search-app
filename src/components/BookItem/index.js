import React from 'react';

const BookItem = ({
  reference,
  title,
  imageLinks: { smallThumbnail },
  description,
}) => {
  return (
    <li className='c-book-item' ref={reference}>
      <img src={smallThumbnail} alt='Book cover' />
      <h3>{title}</h3>
      <p>{description}</p>
    </li>
  );
};

export default BookItem;
