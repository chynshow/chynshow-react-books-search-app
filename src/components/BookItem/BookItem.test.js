import React from 'react';
import BookItem from './';
import { shallow } from 'enzyme';

const setup = (props) => shallow(<BookItem {...props} />);

describe('render BookItem component', () => {
  it('should renders BookItem component with props', () => {
    const c = setup({
      title: 'Title',
      imageLinks: {
        mallThumbnail: '#',
        thumbnail: '#',
      },
      description: 'Description',
      viewType: 'four',
      infoLink: '#',
    });
    expect(c).toMatchSnapshot();
  });

  it('should renders BookItem component without props', () => {
    const c = setup();
    expect(c).toMatchSnapshot();
  });
});
