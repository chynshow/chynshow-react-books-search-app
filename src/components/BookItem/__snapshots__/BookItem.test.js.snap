// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render BookItem component should renders BookItem component with props 1`] = `
<li
  className="c-book-item c-book-item--four"
>
  <img
    alt="Book cover not found"
    className="c-book-item__img"
    src="#"
  />
  <h3
    className="c-book-item__title"
  >
    Title
  </h3>
  <p
    className="c-book-item__description"
  >
    Description...
  </p>
  <a
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    <button
      className="c-btn c-btn--primary c-book-item__btn"
    >
      Details
    </button>
  </a>
</li>
`;

exports[`render BookItem component should renders BookItem component without props 1`] = `
<li
  className="c-book-item c-book-item--four"
>
  <img
    alt="Book cover not found"
    className="c-book-item__img"
    src="#"
  />
  <h3
    className="c-book-item__title"
  >
    Title
  </h3>
  <p
    className="c-book-item__description"
  >
    Description unavailable...
  </p>
  <a
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    <button
      className="c-btn c-btn--primary c-book-item__btn"
    >
      Details
    </button>
  </a>
</li>
`;
